### ISSUE TITLE
webpack 5 dependency is hoisted

### ISSUE THREAD

**Issue by OlenDavis** [nuxt/components#105]
Here is how @nuxt/components currently finds the `vue-loader`'s rule: https://github.com/nuxt/components/blob/bfccd86656d8fc08b4cfad4b8580fd1017ea3349/src/index.ts#L109-L110

Compare that to how vue-loader finds the `vue-loader` rule in the webpack5 version of their plugin: https://github.com/vuejs/vue-loader/blob/e7b2b1181ec2e67db87ba0895c9fb6c5608d5442/lib/plugin-webpack5.js#L49-L60

(And contrast that to how vue-loader finds the `vue-loader` rule in the webpack4 version of their plugin: https://github.com/vuejs/vue-loader/blob/59433190d20bd864d9020168100904791b58cb0d/lib/plugin-webpack4.js#L27-L36)

All that to say there's some work to do for @nuxt/components to support Webpack 5, because on a fresh project using webpack5, users will get the following error using the default `nuxt` (or as in my case, `nuxt-ts`) dev build command:

```
   ╭──────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                  │
   │   ✖ Nuxt Fatal Error                                                                             │
   │                                                                                                  │
   │   Error: Cannot find module 'webpack/lib/RuleSet'                                                │
   │   Require stack:                                                                                 │
   │   - /Users/olen/Development/nuxt-template/node_modules/@nuxt/components/dist/index.js            │
   │   - /Users/olen/Development/nuxt-template/node_modules/@nuxt/core/dist/core.js                   │
   │   - /Users/olen/Development/nuxt-template/node_modules/@nuxt/cli/dist/cli-index.js               │
   │   - /Users/olen/Development/nuxt-template/node_modules/@nuxt/cli/dist/cli.js                     │
   │   - /Users/olen/Development/nuxt-template/node_modules/@nuxt/typescript-runtime/bin/nuxt-ts.js   │
   │                                                                                                  │
   ╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by pi0** [comment]
Hi @OlenDavis Actually you are true. Current version of `@nuxt/components` only works with nuxt v2 which is using WP4. We probably will drop dependency on `RuleSet` for cross version support. 

BTW i think issue is about hoisted version of WP5 which is fixed by [`v1.1.1`](https://github.com/nuxt/components/releases/tag/v1.1.1). (You can upgrade by recreating lock file)

**Comment by CraftingGamerTom** [comment]
I see this was closed. What can we follow for a fix? Ran into these errors when updating libraries. Had to go back to webpack 4 because Nuxt would not start.

**Comment by pi0** [comment]
Dear @CraftingGamerTom, nuxt2 is not (and will not) compatible with webpack5 because of it's breaking changes. You have to wait for nuxt3 released.

**Comment by lsbyerley** [comment]
@pi0 so which version of nuxt do we have to use to fix this error until nuxt3 is released? I don't have webpack in my package json dependencies to downgrade individually

**Comment by pi0** [comment]
@lsbyerley Would you please try with `yarn why webpack`? There might be one sub-dependency depending on webpack5

Anyway i think we should not depend on a side-dependency. Opened to track.

**Comment by lsbyerley** [comment]
@pi0 i don't use yarn, but an `npm ls webpack` results in this..

```
➜ git:(master) ✗ npm ls webpack
└─┬ nuxt@2.14.9
  └─┬ @nuxt/webpack@2.14.9
    └── UNMET PEER DEPENDENCY webpack@4.44.2 

npm ERR! peer dep missing: webpack@^2.0.0 || ^3.0.0 || ^4.0.0, required by @nuxt/friendly-errors-webpack-plugin@2.5.0
npm ERR! peer dep missing: webpack@>=2, required by babel-loader@8.2.2
npm ERR! peer dep missing: webpack@^4.0.0, required by cache-loader@4.1.0
npm ERR! peer dep missing: webpack@^4.0.0 || ^5.0.0, required by css-loader@3.6.0
npm ERR! peer dep missing: webpack@^4.4.0 || ^5.0.0, required by extract-css-chunks-webpack-plugin@4.8.0
npm ERR! peer dep missing: webpack@^4.0.0, required by file-loader@4.3.0
npm ERR! peer dep missing: webpack@*, required by hard-source-webpack-plugin@0.13.1
npm ERR! peer dep missing: webpack@^4.0.0 || ^5.0.0, required by html-webpack-plugin@4.5.0
npm ERR! peer dep missing: webpack@^4.0.0, required by optimize-css-assets-webpack-plugin@5.0.4
npm ERR! peer dep missing: webpack@^3.0.0 || ^4.0.0 || ^5.0.0, required by style-resources-loader@1.4.1
npm ERR! peer dep missing: webpack@^4.0.0 || ^5.0.0, required by terser-webpack-plugin@2.3.8
npm ERR! peer dep missing: webpack@^2.0.0 || ^3.0.0 || ^4.0.0, required by thread-loader@2.1.3
npm ERR! peer dep missing: webpack@>=4.0.0, required by time-fix-plugin@2.0.7
npm ERR! peer dep missing: webpack@^4.0.0, required by url-loader@2.3.0
npm ERR! peer dep missing: webpack@^3.0.0 || ^4.1.0 || ^5.0.0-0, required by vue-loader@15.9.5
npm ERR! peer dep missing: webpack@^4.0.0, required by webpack-dev-middleware@3.7.2
npm ERR! peer dep missing: webpack@^3.0.0 || ^4.0.0, required by webpackbar@4.0.0
```

**Comment by pi0** [comment]
Issue should be fixed in [v1.2.1](https://github.com/nuxt/components/releases/tag/v1.2.1) by #119. You can get update using `yarn upgeade @nuxt/components` or `npm up @nuxt/components`. Please reopen if still not working.

---

@lsbyerley I think issue of hosting was either with `hard-source-webpack-plugin` or `time-fix-plugin` which both have a peerDependency range that allows webpack 5 and npm7 installs peerDependencies by default. We are no longer depending on webpack as peer dependency.
